pkg/sql/parser/sql.y

```diff
@@ -8124,6 +8124,11 @@ sequence_option_elem:
   AS typename                  {
                                   // Valid option values must be integer types (ex. int2, bigint)
                                   parsedType := $2.colType()
+                                  if parsedType == nil {
+                                      sqllex.(*lexer).lastError = pgerror.Newf(pgcode.UndefinedObject, "type %q does not exist", $2.val)
```

Comment:
i think "UndefinedObject" may not be right here. I feel like we'd want it to match the error below on line 8133

Reply:
This is the error code PG uses in this case:

```
marcus=# create sequence s1 as abc;
ERROR:  42704: type "abc" does not exist
LINE 1: create sequence s1 as abc;
```

But if you think it's ok to diverge in this case then I'm fine with the standard parsing error code.

Reply:
Should I close this since #82213 is approved?